src/zenml/zen_stores/schemas/step_run_schemas.py

```diff
@@ -212,11 +213,27 @@ def to_model(
             for artifact in self.output_artifacts
         }
 
+        # the step configuration moved into the deployment - the else case is to
+        # guarantee backwards compatibility
         if self.deployment is not None:
-            full_step_config = Step.model_validate(
-                json.loads(self.deployment.step_configurations)[self.name]
+            step_configuration = json.loads(
+                self.deployment.step_configurations
             )
-        elif self.step_configuration is not None:
+            if self.name in step_configuration:
+                full_step_config = Step.model_validate(
+                    step_configuration[self.name]
+                )
+            elif self.step_configuration:
```

Comment:
Isn't this case already handled below?

Maybe 
```python
  ...
elif not self.step_configuration:
  raise ValueError(...)
```

Reply:
but it won't ever end up in the elif below once it goes into the else statement

Reply:
My suggestion is to get rid of the `else` end use an `elif` instead, see the code above

Reply:
```python
if self.name in step_configuration:
    full_step_config = Step.model_validate(
        step_configuration[self.name]
    )
elif not self.step_configuration:
    raise ValueError(...)
```